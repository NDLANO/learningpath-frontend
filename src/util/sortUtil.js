/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import sortBy from 'lodash/sortBy';
import reverse from 'lodash/reverse';

export const sortPaths = (paths, field) => {
  switch (field) {
    case 'title':
      return sortBy(paths, p => p.title);

    case 'lastUpdated':
      return sortBy(paths, field);

    case '-lastUpdated':
      return reverse(sortBy(paths, 'lastUpdated'));

    case 'status':
      return reverse(sortBy(paths, 'status'));

    default:
      return sortBy(paths, field);
  }
};
