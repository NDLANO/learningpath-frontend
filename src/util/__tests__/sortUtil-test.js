/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { sortPaths } from '../sortUtil';
import { translatedLearningPaths } from '../../common/__tests__/translatedMockData';

test('util/sortUtil sortPaths', () => {
  expect(sortPaths(translatedLearningPaths, 'status')).toMatchSnapshot();
  expect(sortPaths(translatedLearningPaths, 'title')).toMatchSnapshot();
  expect(sortPaths(translatedLearningPaths, 'lastUpdated')).toMatchSnapshot();
  expect(sortPaths(translatedLearningPaths, '-lastUpdated')).toMatchSnapshot();
  expect(sortPaths(translatedLearningPaths, 'duration')).toMatchSnapshot();
});
