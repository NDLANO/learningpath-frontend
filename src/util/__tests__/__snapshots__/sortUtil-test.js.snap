// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/sortUtil sortPaths 1`] = `
Array [
  Object {
    "author": Object {
      "name": "Snurre Sprett",
      "type": "writer",
    },
    "description": "Kurset dekker innføring og vil gi deg grunnleggende forståelse for vanlige begrep i kunst og kultur verden. Kurset fokuserer på kunst og kultur på et verdensperspektiv.",
    "duration": 1080,
    "id": "1",
    "introduction": "<p>Norsk test id 1</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2015-03-07T13:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/1",
    "status": "PUBLISHED",
    "tags": Array [
      "kristofer",
    ],
    "title": "Hva er kunst og kultur?",
  },
  Object {
    "author": Object {
      "name": "Kaptein Sabeltann",
      "type": "writer",
    },
    "description": "Uttrykkene \\"leselighet\\" og \\"lesbarhet\\" brukes om hverandre i norsk fagterminologi, og ofte uten klare forestillinger om hva begrepene står for.",
    "duration": 45,
    "id": "2",
    "introduction": "<p>Norsk test</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2016-02-01T11:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/2",
    "status": "PRIVATE",
    "tags": Array [
      "kristofer",
    ],
    "title": "Leselighet og skrift",
  },
]
`;

exports[`util/sortUtil sortPaths 2`] = `
Array [
  Object {
    "author": Object {
      "name": "Snurre Sprett",
      "type": "writer",
    },
    "description": "Kurset dekker innføring og vil gi deg grunnleggende forståelse for vanlige begrep i kunst og kultur verden. Kurset fokuserer på kunst og kultur på et verdensperspektiv.",
    "duration": 1080,
    "id": "1",
    "introduction": "<p>Norsk test id 1</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2015-03-07T13:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/1",
    "status": "PUBLISHED",
    "tags": Array [
      "kristofer",
    ],
    "title": "Hva er kunst og kultur?",
  },
  Object {
    "author": Object {
      "name": "Kaptein Sabeltann",
      "type": "writer",
    },
    "description": "Uttrykkene \\"leselighet\\" og \\"lesbarhet\\" brukes om hverandre i norsk fagterminologi, og ofte uten klare forestillinger om hva begrepene står for.",
    "duration": 45,
    "id": "2",
    "introduction": "<p>Norsk test</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2016-02-01T11:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/2",
    "status": "PRIVATE",
    "tags": Array [
      "kristofer",
    ],
    "title": "Leselighet og skrift",
  },
]
`;

exports[`util/sortUtil sortPaths 3`] = `
Array [
  Object {
    "author": Object {
      "name": "Snurre Sprett",
      "type": "writer",
    },
    "description": "Kurset dekker innføring og vil gi deg grunnleggende forståelse for vanlige begrep i kunst og kultur verden. Kurset fokuserer på kunst og kultur på et verdensperspektiv.",
    "duration": 1080,
    "id": "1",
    "introduction": "<p>Norsk test id 1</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2015-03-07T13:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/1",
    "status": "PUBLISHED",
    "tags": Array [
      "kristofer",
    ],
    "title": "Hva er kunst og kultur?",
  },
  Object {
    "author": Object {
      "name": "Kaptein Sabeltann",
      "type": "writer",
    },
    "description": "Uttrykkene \\"leselighet\\" og \\"lesbarhet\\" brukes om hverandre i norsk fagterminologi, og ofte uten klare forestillinger om hva begrepene står for.",
    "duration": 45,
    "id": "2",
    "introduction": "<p>Norsk test</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2016-02-01T11:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/2",
    "status": "PRIVATE",
    "tags": Array [
      "kristofer",
    ],
    "title": "Leselighet og skrift",
  },
]
`;

exports[`util/sortUtil sortPaths 4`] = `
Array [
  Object {
    "author": Object {
      "name": "Kaptein Sabeltann",
      "type": "writer",
    },
    "description": "Uttrykkene \\"leselighet\\" og \\"lesbarhet\\" brukes om hverandre i norsk fagterminologi, og ofte uten klare forestillinger om hva begrepene står for.",
    "duration": 45,
    "id": "2",
    "introduction": "<p>Norsk test</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2016-02-01T11:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/2",
    "status": "PRIVATE",
    "tags": Array [
      "kristofer",
    ],
    "title": "Leselighet og skrift",
  },
  Object {
    "author": Object {
      "name": "Snurre Sprett",
      "type": "writer",
    },
    "description": "Kurset dekker innføring og vil gi deg grunnleggende forståelse for vanlige begrep i kunst og kultur verden. Kurset fokuserer på kunst og kultur på et verdensperspektiv.",
    "duration": 1080,
    "id": "1",
    "introduction": "<p>Norsk test id 1</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2015-03-07T13:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/1",
    "status": "PUBLISHED",
    "tags": Array [
      "kristofer",
    ],
    "title": "Hva er kunst og kultur?",
  },
]
`;

exports[`util/sortUtil sortPaths 5`] = `
Array [
  Object {
    "author": Object {
      "name": "Kaptein Sabeltann",
      "type": "writer",
    },
    "description": "Uttrykkene \\"leselighet\\" og \\"lesbarhet\\" brukes om hverandre i norsk fagterminologi, og ofte uten klare forestillinger om hva begrepene står for.",
    "duration": 45,
    "id": "2",
    "introduction": "<p>Norsk test</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2016-02-01T11:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/2",
    "status": "PRIVATE",
    "tags": Array [
      "kristofer",
    ],
    "title": "Leselighet og skrift",
  },
  Object {
    "author": Object {
      "name": "Snurre Sprett",
      "type": "writer",
    },
    "description": "Kurset dekker innføring og vil gi deg grunnleggende forståelse for vanlige begrep i kunst og kultur verden. Kurset fokuserer på kunst og kultur på et verdensperspektiv.",
    "duration": 1080,
    "id": "1",
    "introduction": "<p>Norsk test id 1</p>",
    "isBasedOnTitle": undefined,
    "lastUpdated": 2015-03-07T13:00:00.000Z,
    "metaUrl": "http://api.test.ndla.no/learningpaths/1",
    "status": "PUBLISHED",
    "tags": Array [
      "kristofer",
    ],
    "title": "Hva er kunst og kultur?",
  },
]
`;
